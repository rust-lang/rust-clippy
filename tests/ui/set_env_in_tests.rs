#![warn(clippy::set_env_in_tests)]

use std::env;

fn main() {
    unsafe { env::set_var("CLIPPY_TESTS_THIS_IS_OK", "1") }
}

#[cfg(test)]
mod tests {
    use std::env;

    #[test]
    fn my_test() {
        unsafe { env::set_var("CLIPPY_TESTS_THIS_IS_NOT_OK", "1") }
        //~^ set_env_in_tests
    }
}
