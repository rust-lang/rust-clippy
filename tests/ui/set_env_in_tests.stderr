error: env::set_var called from a test
  --> tests/ui/set_env_in_tests.rs:15:18
   |
LL |         unsafe { env::set_var("CLIPPY_TESTS_THIS_IS_NOT_OK", "1") }
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: this might indicate state leakage and flaky tests
   = note: `-D clippy::set-env-in-tests` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::set_env_in_tests)]`

error: aborting due to 1 previous error

